### ISSUE TITLE
Migrate notifications/user_settings_notifications.jsx to typescript

### ISSUE THREAD

**Issue by johanmoritz** [adbjo/mattermost-webapp#2]
`NotificationsTab` is the default export.
 
It has the implicit prop type:

```javascript
static propTypes = {
        user: PropTypes.object,
        updateSection: PropTypes.func,
        activeSection: PropTypes.string,
        closeModal: PropTypes.func.isRequired,
        collapseModal: PropTypes.func.isRequired,
        sendPushNotifications: PropTypes.bool,
        enableAutoResponder: PropTypes.bool,
        actions: PropTypes.shape({
            updateMe: PropTypes.func.isRequired,
        }).isRequired,
    }
```

with defaults

```javascript
static defaultProps = {
        user: null,
        activeSection: '',
        activeTab: '',
    }
```

and internal state

``` typescript
interface State {
  enableEmail: 'true' | ??,
  desktopSound: 'true' | ??,
  desktopActivity: NotificationLevels,
  pushActivity: NotificationLevels,
  pushStatus: Constants.UserStatuses,
  notifyCommentsLevel: 'never' | ??,
  autoResponderActive: boolean,
  autoResponderMessage: ReturnType<Utils.localizeMessage>; // string?
  usernameKey: boolean,
  customKeys: string,
  customKeysChecked: boolean,
  firstNameKey: boolean,
  channelKey: boolean,
  serverError: ??,
  isSaving: boolean,
}
```

We have a User in propTypes that looks like: 

```typescript
interface User {
  notify_props?: {
    desktop?: ??,
    desktop_sound?: ??,
    comments?: ??,
    email?: ??,
    push?: ??,
    push_status?: ??,
    auto_responder_active?: 'true' | ??,
    auto_responder_message?: ??
    mention_keys?: string;
  }
}
```

We can create possible Props as:

```typescript
interface Props {
  user?: User;
  updateSection: (??) => ??;
  activeSection: string;
  closeModal: (??) => ??;
  collapseModal: (??) => ??;
  sendPushNotifications: boolean;
  enableAutoResponder: boolean;
  actions: {
    updateMe: (??) => ??;
  };
}
```

**Comment by johanmoritz** [comment]
Hard changes: 

* `user` must be defined (cannot be undefined/null)

Questions: 
* `auto_reponder_active` is not defined in `UserNotifyProps`, can it be safely removed?
* `auto_responder_message` is not defined in `UserNotifyProps`, can it be safely removed?

**Comment by adbjo** [comment]
Solved in 9bd62b773f12a91efe95f8dd3123c04cd65279c5
